packages/repository-tests/src/crud/replace-by-id.suite.ts

```diff
@@ -71,7 +76,9 @@ export function createSuiteForReplaceById(
       await repo.replaceById(created.id, created);
 
       const found = await repo.findById(created.id);
-      expect(toJSON(found)).to.deepEqual(
+      // use containDeep instead of deepEqual because cloudant changes _rev value after
+      // replacement
+      expect(toJSON(found)).to.containDeep(
```

Comment:
I am concerned that by changing from `deepEqual` to `containDeep` we are allowing many more changes than just a different `_rev` value. For example, if the connector and/or the database adds new properties, this assertion will happily accept that, even though such behavior is most likely not wanted.

Let's find a way how to deal with `_rev` changes via a new `CrudFeatures` flag in such way that only changes in the `_rev` property is allowed and only when testing against Cloudant/CouchDB.

Reply:
Updated in the fifth commit.